jupyter_server/serverapp.py

```diff
@@ -1951,6 +1953,19 @@ def init_logging(self):
     def init_event_logger(self):
         """Initialize the Event Bus."""
         self.event_logger = EventLogger(parent=self)
+        # Load the core Jupyter Server event schemas
+        # All event schemas must start with Jupyter Server's
+        # events URI, `JUPYTER_SERVER_EVENTS_URI`.
+        schema_ids = [
+            "https://events.jupyter.org/jupyter_server/contents_service/v1",
+        ]
+        for schema_id in schema_ids:
+            # Get the schema path from the schema ID.
+            rel_schema_path = schema_id.lstrip(JUPYTER_SERVER_EVENTS_URI) + ".yaml"
+            schema_path = DEFAULT_EVENTS_SCHEMA_PATH / rel_schema_path
+            # Use this pathlib object to register the schema
+            # breakpoint()
+            self.event_logger.register_event_schema(schema_path)
```

Comment:
Does `schema_path` need to be a `Pathlib` object? This doesn't appear documented, but when I tried passing in a string, I got this error:

```
dev-dsk-dlq-2b-55651109 % jupyter server --ip=0.0.0.0
Traceback (most recent call last):
  File "/home/dlq/anaconda3/envs/jserver-dev/bin/jupyter-server", line 8, in <module>
    sys.exit(main())
  File "/home/dlq/anaconda3/envs/jserver-dev/lib/python3.9/site-packages/jupyter_core/application.py", line 269, in launch_instance
    return super().launch_instance(argv=argv, **kwargs)
  File "/home/dlq/anaconda3/envs/jserver-dev/lib/python3.9/site-packages/traitlets/config/application.py", line 975, in launch_instance
    app.initialize(argv)
  File "/home/dlq/anaconda3/envs/jserver-dev/lib/python3.9/site-packages/traitlets/config/application.py", line 110, in inner
    return method(app, *args, **kwargs)
  File "/workplace/dlq/jupyter_server/jupyter_server/serverapp.py", line 2495, in initialize
    self.init_event_logger()
  File "/workplace/dlq/jupyter_server/jupyter_server/serverapp.py", line 1958, in init_event_logger
    self.event_logger.register_event_schema(event_schema_path)
  File "/home/dlq/anaconda3/envs/jserver-dev/lib/python3.9/site-packages/jupyter_events/logger.py", line 134, in register_event_schema
    event_schema = self.schemas.register(schema)
  File "/home/dlq/anaconda3/envs/jserver-dev/lib/python3.9/site-packages/jupyter_events/schema_registry.py", line 42, in register
    schema = EventSchema(schema)
  File "/home/dlq/anaconda3/envs/jserver-dev/lib/python3.9/site-packages/jupyter_events/schema.py", line 44, in __init__
    _schema = self._load_schema(schema)
  File "/home/dlq/anaconda3/envs/jserver-dev/lib/python3.9/site-packages/jupyter_events/schema.py", line 66, in _load_schema
    raise EventSchemaLoadingError(
jupyter_events.schema.EventSchemaLoadingError: When deserializing `schema`, we expected a dictionary to be returned but a <class 'str'> was returned instead. Double check the `schema` to make sure it is in the proper form.
```

For reference, here was how I attempted to register the schema:

```py
    def init_event_logger(self):
        """Initialize the Event Bus."""
        self.event_logger = EventLogger(parent=self)
        event_schema_paths = [
            "jupyter_server/event_schemas/contents.json" # [sic]
        ]
        for event_schema_path in event_schema_paths:
            self.event_logger.register_event_schema(event_schema_path)
```

Replacing `"jupyter_server/event_schemas/contents.json"` with `pathlib.Path("jupyter_server/event_schemas/contents.json")` makes the error go away.

This error also shows up if you give an EventLogger a non-existent path like `asdf.json`. Perhaps a sign that better schema path validation should be employed in the body of `register_event_schema()`.
```

Reply:
This is actually a consequence of https://github.com/jupyter/jupyter_events/pull/15

We found that interpreting a string as a path is **super** insecure. This is a known issue with Python path interpolation, so it's _always_ best to use an explicit pathlib object when reading a file path. This is a large reason why the pathlib library was created in Python. 

I agree that more documentation—and perhaps validation—on the jupyter_events side might be required. If you're up for it, do you mind creating an issue over there, we can solve this in jupyter_events directly. I'm happy to open the issue too, if you'd rather. Thanks!

Reply:
Yeah. I think we just need to update the docs and the examples within it (since that's why I tried using a string for path in the first place). I'll open a PR for that.

Reply:
Ah, another PR also should be added for improving the error message raised in `schema.py`.

Reply:
Sounds good. Thanks @dlqqq!